pkg/httputils/http_utils_test.go

```diff
@@ -2,24 +2,46 @@ package httputils
 
 import (
 	"net/http"
+	"strings"
 	"testing"
 )
 
 func TestBoolValue(t *testing.T) {
-	type args struct {
-		r *http.Request
-		k string
-	}
 	tests := []struct {
-		name string
-		args args
-		want bool
+		name    string
+		testStr string
+		want    bool
 	}{
-		// TODO: Add test cases.
+		{"test normal string", "normal string", true},
+		{"test ''", "", false},
+		{"test 0", "0", false},
+		{"test no", "no", false},
+		{"test NO", "NO", false},
+		{"test false", "false", false},
+		{"test FALSE", "FALSE", false},
+		{"test none", "none", false},
+		{"test NONE", "NONE", false},
+		// test trim
+		{"test '  '", "  ", false},
+		{"test trim 0", "  0  ", false},
+		{"test trim no", "  no  ", false},
+		{"test trim NO", "  NO ", false},
+		{"test trim false", " false ", false},
+		{"test trim FALSE", "  FALSE ", false},
+		{"test trim none", "  none ", false},
+		{"test trim NONE", "  NONE  ", false},
 	}
+	boolName := "name"
 	for _, tt := range tests {
+		request, _ := http.NewRequest(
+			"POST",
+			"http://test",
+			strings.NewReader(boolName+"="+tt.testStr))
+
+		request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
 		t.Run(tt.name, func(t *testing.T) {
-			if got := BoolValue(tt.args.r, tt.args.k); got != tt.want {
+			if got := BoolValue(request, boolName); got != tt.want {
 				t.Errorf("BoolValue() = %v, want %v", got, tt.want)
```

Comment:
maybe we could print tt.name to help us find which testcase is wrong?

Reply:
it seems that t.Run(tt.name,...) will notify us which testcase is wrong like this:
![image](https://user-images.githubusercontent.com/12333609/50087572-7291e200-023b-11e9-9a19-89105aa95a7d.png)

Reply:
@ZYecho I've added it